listenbrainz/webserver/static/js/src/APIService.ts

```diff
@@ -161,31 +161,33 @@ export default class APIService {
 
       const url = `${this.APIBaseURI}/submit-listens`;
 
-      try {
-        const response = await fetch(url, {
-          method: "POST",
-          headers: {
-            Authorization: `Token ${userToken}`,
-            "Content-Type": "application/json;charset=UTF-8",
-          },
-          body: JSON.stringify(struct),
-        });
-
-        // we skip listens if we get an error code that's not a rate limit
-        if (response.status === 429) {
-          // Rate limit error, this should never happen, but if it does, try again in 3 seconds.
+      while(true){
+        try {
+          const response = await fetch(url, {
+            method: "POST",
+            headers: {
+              Authorization: `Token ${userToken}`,
+              "Content-Type": "application/json;charset=UTF-8",
+            },
+            body: JSON.stringify(struct),
+          });
+  
+          // we skip listens if we get an error code that's not a rate limit
+          if (response.status === 429) {
+            // Rate limit error, this should never happen, but if it does, try again in 3 seconds.
+            setTimeout(
```

Comment:
This won't cause the JavaScript to sleep for 3 seconds here, it will simply setup a time to call the given callback after 3 seconds and continue. If you want it to sleep you can wait on a `Promise` which gets resolved in 3 seconds like this.
```js
await new Promise((resolve) => setTimeout(resolve, 3000))
```